bob_the_builder/src/cargo_toml.rs

```diff
@@ -1,92 +1,222 @@
+                    Ok(IsWorkspace::Yes { members })
+                } else {
+                    Ok(IsWorkspace::NoMembers)
+                }
             } else {
                 Ok(IsWorkspace::NoMembers)
             }
         } else {
-            Ok(IsWorkspace::NoMembers)
+            Ok(IsWorkspace::No)
         }
-    } else {
-        Ok(IsWorkspace::No)
     }
-}
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+    #[cfg(test)]
+    mod tests {
+        use super::*;
 
-    #[test]
-    fn is_workspace_works() {
-        let is = is_workspace(
-            r#"
+        #[test]
+        fn is_workspace_works() {
+            let is = is_workspace(
+                r#"
             title = 'TOML Example'
 
             [workspace]
             "#,
-        )
-        .unwrap();
-        assert_eq!(is, IsWorkspace::NoMembers);
-        let is = is_workspace(
-            r#"
+            )
+            .unwrap();
+            assert_eq!(is, IsWorkspace::NoMembers);
+            let is = is_workspace(
+                r#"
             title = 'TOML Example'
 
             [workspace]
             members = []
             "#,
-        )
-        .unwrap();
-        assert_eq!(is, IsWorkspace::NoMembers);
+            )
+            .unwrap();
+            assert_eq!(is, IsWorkspace::NoMembers);
 
-        let is = is_workspace(
-            r#"
+            let is = is_workspace(
+                r#"
             title = 'TOML Example'
             "#,
-        )
-        .unwrap();
-        assert_eq!(is, IsWorkspace::No);
+            )
+            .unwrap();
+            assert_eq!(is, IsWorkspace::No);
 
-        let is = is_workspace(
-            r#"
+            let is = is_workspace(
+                r#"
             title = 'TOML Example'
 
             [workspace]
             members = ["contracts/*"]
             "#,
-        )
-        .unwrap();
-        assert_eq!(
-            is,
-            IsWorkspace::Yes {
-                members: vec!["contracts/*".to_string()]
-            }
-        );
+            )
+            .unwrap();
+            assert_eq!(
+                is,
+                IsWorkspace::Yes {
+                    members: vec!["contracts/*".to_string()]
+                }
+            );
+        }
+    }
+}
+
+pub mod package {
+    use std::hash::Hash;
+
+    use serde::Deserialize;
+
+    use crate::pkg_build::ParsedPackage;
+
+    pub type BuildName = String;
+    pub type Feature = String;
+
+    #[derive(Deserialize, Debug)]
+    pub struct PackageCargoToml {
+        package: Package,
+    }
+
+    #[derive(Deserialize, Debug)]
+    pub struct Package {
+        name: String,
+        metadata: Option<OptimizerMetadata>,
+    }
+
+    #[derive(Deserialize, Debug)]
+    pub struct OptimizerMetadata {
+        optimizer: Option<Optimizer>,
+    }
+
+    #[derive(Deserialize, Debug, Default)]
+    pub struct Optimizer {
+        #[serde(rename = "default-build")]
```

Comment:
Should we go for `#[serde(rename_all = "kebab-case")]` instead to ensure consistency?